### PR TITLE
Tracking editor events

### PR DESCRIPTION
**Pull request by ezequielc**
@alex-deaconu Please review. Cheers

### CODE REVIEW

File: js/services/svc-editor-factory.js [hunk 1]
```diff
@@ -309,12 +309,13 @@ angular.module('risevision.editorApp.services')
...
       };
 
-      factory.newCopyOf = function(presentationId){
+      factory.newCopyOf = function (presentationId) {
         factory.getPresentation(presentationId)
           .then(factory.copyPresentation);
       };
 
       factory.addFromTemplate = function () {
+        presentationTracker('Add Presentation From Template');
```
Comment (alex-deaconu) [inline js/services/svc-editor-factory.js]: @ezequielc wouldn't it make more sense to track this once the user has actually selected a template? What if they cancel the action? Thanks.

File: js/directives/dtv-toolbar.js [hunk 2]
```diff
@@ -10,6 +11,8 @@ angular.module('risevision.editorApp.directives')
             $scope.previewUrl = editorFactory.getPreviewUrl();
 
             $scope.addNewPlaceholder = function () {
+              presentationTracker('Add Placeholder', editorFactory.presentation
+                .id, editorFactory.presentation.name);
```
Comment (alex-deaconu) [inline js/directives/dtv-toolbar.js]: @ezequielc as per the pattern of your other tracking events, shouldn't this go in the `placeholdersFactory`? Thanks.

File: js/services/svc-playlist-item-factory.js [hunk 3]
```diff
@@ -42,6 +43,8 @@ angular.module('risevision.editorApp.services')
       };
 
       factory.addContent = function () {
+        presentationTracker('Add Content', editorFactory.presentation.id,
+          editorFactory.presentation.name);
```
Comment (alex-deaconu) [inline js/services/svc-playlist-item-factory.js]: @ezequielc shouldn't this event be tracked after the user selects content? What if they cancel their action? Thanks.

File: partials/placeholder-playlist.html [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <div dropdown>
-  <button id="addPlaylistItemButton" dropdown-toggle class="dropdown-toggle btn btn-primary btn-block" >{{'editor-app.playlistItem.add' | translate}}  <i class="fa fa-plus icon-right"></i></button>
+  <button id="addPlaylistItemButton" dropdown-toggle ng-click="presentationTracker('Add Playlist Item')" class="dropdown-toggle btn btn-primary btn-block" >{{'editor-app.playlistItem.add' | translate}}  <i class="fa fa-plus icon-right"></i></button>
```
Comment (alex-deaconu) [inline partials/placeholder-playlist.html]: @ezequielc is this tracking a user click? Or the action of adding a playlist item? To confirm, this click merely opens a dropdown menu which the user can just choose to close: http://screencast.com/t/HuidCKOGCYoW (generating useless events). Thanks.

File: js/directives/dtv-toolbar.js (hunk 2 above)
Comment (ezequielc) [inline js/directives/dtv-toolbar.js]: @alex-deaconu Good point. I will change it. Cheers

File: js/services/svc-editor-factory.js (hunk 1 above)
Comment (ezequielc) [inline js/services/svc-editor-factory.js]: @alex-deaconu I think we should actually track both events. What do you think? Cheers

File: partials/presentation-list.html [hunk 5]
```diff
@@ -8,11 +8,11 @@ <h1 id="title" translate>editor-app.list.title</h1>
     <div class="global-actions">
 
       <div class="btn-group" dropdown>
-        <button id="presentationAddButton" dropdown-toggle class="dropdown-toggle btn btn-primary btn-lg">{{'editor-app.list.actions.add' |translate}} <i class="fa fa-plus icon-right"></i></button>
+        <button id="presentationAddButton" ng-click="presentationTracker('Add Presentation')" dropdown-toggle class="dropdown-toggle btn btn-primary btn-lg">{{'editor-app.list.actions.add' |translate}} <i class="fa fa-plus icon-right"></i></button>
```
Comment (alex-deaconu) [inline partials/presentation-list.html]: @ezequielc is this event tracking a user click? Or the action of adding a presentation? To confirm, this click merely opens a dropdown menu: http://screencast.com/t/LV15DwCK7So. Thanks.

File: js/services/svc-playlist-item-factory.js (hunk 3 above)
Comment (ezequielc) [inline js/services/svc-playlist-item-factory.js]: @alex-deaconu Same as previous comment. I think we should track both moments. Cheers

File: js/services/svc-editor-factory.js (hunk 1 above)
Comment (alex-deaconu) [inline js/services/svc-editor-factory.js]: @ezequielc I'm not sure, depends on the information you need to obtain. However, I would say that tracking too many events has its downsides as well. Thanks.

File: partials/presentation-list.html (hunk 5 above)
Comment (ezequielc) [inline partials/presentation-list.html]: @alex-deaconu Yes, we are tracking dropdown clicks in both cases. 
@ghislaineguerin Should we keep it or remove it? Thanks

Comment (alex-deaconu) [inline partials/presentation-list.html]: @ezequielc okay, thanks for the confirmation.

File: js/services/svc-editor-factory.js (hunk 1 above)
Comment (ezequielc) [inline js/services/svc-editor-factory.js]: @alex-deaconu I agree with you. But I think at this stage it is better to have a more detailed information. I would add both for now and in the future we should remove some of the events. What do you think? Cheers

Comment (alex-deaconu) [inline js/services/svc-editor-factory.js]: Sounds good. Thanks.

**Comment by ezequielc** [comment]
@alex-deaconu I added events when selecting from Store and moved the code from toolbar directive to the factory. Please review. Thanks!

**Comment by alex-deaconu** [comment]
Looks good. Thanks for fixing :+1: .

**Pull request merged**